### ISSUE TITLE
'autolink' doesn't work

### ISSUE THREAD

**Issue by naokazuterada** [naokazuterada/MarkdownTOC#109]
```markdown
<!-- MarkdownTOC autolink="true" bracket="round" -->

- [variable \](#required\)

<!-- /MarkdownTOC -->


# variable \[required\]
```

should be

```markdown
<!-- MarkdownTOC autolink="true" bracket="round" -->

- [variable\[#required\]](#variable-required)

<!-- /MarkdownTOC -->


# variable \[required\]
```

**Comment by naokazuterada** [comment]
On GFM, `[`and`]` seems to be  escaped always, but this is conflicting Markdown specification maybe.

**Comment by naokazuterada** [comment]
**Idea**
- Separate GFM mode and Original Markdown mode
    - including `autolink=true` and `bracket=round`

**Comment by ZimbiX** [comment]
Headings containing *rendered underscores* are also affected by this issue. That is, the anchors when displayed on GitHub do not match.

Test GFM:

```
<!-- MarkdownTOC depth=0 autolink=true bracket=round -->

<!-- /MarkdownTOC -->

# 1 test_x

# 2 test _x_

# 3 test _x

# 4 test x_
```

Expected (GitHub anchors):

```
- [1 test_x](#1-test_x)
- [2 test _x_](#2-test-x)
- [3 test _x](#3-test-_x)
- [4 test x_](#4-test-x_)
```

Actual:

```
- [1 test_x](#1-testx)
- [2 test _x_](#2-test-x)
- [3 test _x](#3-test-x)
- [4 test x_](#4-test-x)
```

Removing `_` from `id_replacements` doesn't fix the issue:

```
- [1 test_x](#1-test_x)
- [2 test _x_](#2-test-_x_)
- [3 test _x](#3-test-_x)
- [4 test x_](#4-test-x_)
```

**Comment by naokazuterada** [comment]
@ZimbiX 
Thank you for reporting. I will take a look 👀

**Comment by naokazuterada** [comment]
Fixed in [3.0.0](https://github.com/naokazuterada/MarkdownTOC/releases/tag/3.0.0).
